### PR TITLE
A0-2406: Handle incoming blocks

### PR DESCRIPTION
**Pull request by lesniak43**
# Description

Puts the incoming blocks into Substrate import queue.

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

### CODE REVIEW

File: finality-aleph/src/sync/service.rs [hunk 1]
```diff
@@ -282,10 +289,11 @@ where
...
                 let state = request.state().clone();
                 self.handle_request(request, peer.clone());
-                self.handle_state(state, peer)
+                self.handle_state(state, peer);
             }
-            RequestResponse(_, justifications) => {
-                self.handle_justifications(justifications, Some(peer))
+            RequestResponse(blocks, justifications) => {
+                self.handle_justifications(justifications, Some(peer));
+                self.handle_blocks(blocks);
```
Comment (kostekIV) [inline finality-aleph/src/sync/service.rs:296]: hmmm, shouldnt we first import block and only then handle the justifications? Also its quite confusing to me why the response has both blocks and justifications in body. Shouldn't it be the separate response for justifications and blocks?

File: finality-aleph/src/sync/handler.rs [hunk 2]
```diff
@@ -250,6 +250,11 @@ where
         Ok(maybe_id)
     }
 
+    /// Handle a single block.
+    pub fn handle_block(&mut self, block: B) {
+        self.block_importer.import_block(block)
```
Comment (timorl) [inline finality-aleph/src/sync/handler.rs:255]: Uhhh, you don't want to just import any blocks that someone throws at you, you gotta check whether they are required, no?

File: finality-aleph/src/sync/service.rs (hunk 1 above)
Comment (timorl) [reply finality-aleph/src/sync/service.rs:296]: The order doesn't matter that much, as handling justifications only puts them in the forest, they get actually justified when we get info that blocks got imported.

**Review by timorl — CHANGES_REQUESTED** [review]
I think you missed a very important part of the logic here. <_<

File: finality-aleph/src/sync/handler.rs (hunk 2 above)
Comment (lesniak43) [reply finality-aleph/src/sync/handler.rs]: immensely reasonable, fixed

File: finality-aleph/src/sync/service.rs (hunk 1 above)
Comment (lesniak43) [reply finality-aleph/src/sync/service.rs]: also, the main idea is that the response represents "all what can be useful to us according to the peer, based on info from our request", hence single message variant

Comment (kostekIV) [reply finality-aleph/src/sync/service.rs]: Imho its better for code readability to make distinction here (in enum variant). Anyway resolving, looks good as it is :)

**Review by kostekIV — APPROVED** [review]

**Pull request merged**